src/main/java/com/twilio/http/Request.java

```diff
@@ -48,6 +51,13 @@ public Request(final HttpMethod method, final String domain, final String uri, f
     public void setAuth(final String username, final String password) {
         this.username = username;
         this.password = password;
+        this.authStrategy = null;
+    }
+
+    public void setAuth(final AuthStrategy authStrategy) {
```

Comment:
Please add a null check here for auth strategy, otherwise even a null authstrategy might clear username and password